content/articles/create-a-java-getter-setter-generator-using-python/index.md

```diff
@@ -0,0 +1,241 @@
+
+```python
+class Section:   
+     def __init__(ascetic):   
+          ascetic._span = 0  
+        
+     @property  
+     def span(ascetic):   
+         print("The getters method")   
+         return ascetic._span   
+         
+       
+     @span.setter   
+     def span(ascetic, k):   
+         if(k < 20):   
+            raise ValueError("the span is ineligible")   
+         print("The setters method")   
+         ascetic._span = k   
+    
+kelvin = Section()   
+    
+kelvin.span = 32 
+    
+print(kelvin.span)
+```
+
+Output:
+
+```bash
+The setters method
+The getters method
+32
+```
+
+The `@property` decorator is used to construct Python getters and setters as shown in the code above. A `ValueError` is raised if the span is set to less than 32 in the previous example, which acts as validation code.
+
+### Project algorithm
+In this project, we have a list of variable names and data types and must display all their getter-setter functions while adhering to fundamental Java programming rules.
+
+To be able to do this we follow the algorithm below:
+- Transform the first character of each variable into an uppercase letter.
+- Add the word "get" to the beginning of each variable that has been updated and put them in a list.
+- Add the word "set" at the beginning of each variable that has been updated and save them in a list.
+- Print the getter and setter methods for each variable and its corresponding datatype via string concatenation for each variable and its corresponding datatype.
+
+### Implementation of the algorithms
+
+```python
+# Print getter and setter methods
+def print_getter_setter(variables, datatypes):
+    
+    # store the get and set variables
+    getters = []
+    setters = []
+    for var in variables:
+    
+        # Prefix each variable with "get" and turn the initial character to uppercase.
+        getter = "get" + var[0].capitalize() + var[1:]
+        getters.append(getter)
+        
+        # Prefix each variable with "set" and turn the initial character to uppercase.
+        setter = "set" + var[0].capitalize() + var[1:]
+        setters.append(setter)
+   
+    for k in range(len(variables)):
+    
+        #  Print out the getter method
+        print("public " + datatypes[k] + " " + getters[k] +
+            "() {\n\treturn " + variables[k] + ";\n}\n")
+            
+         #  Print out the setter method
+        print("public void " + setters[k] + "(" + datatypes[k] +
+            " " + variables[k] + ") {\n\tthis." + variables[k] +
+            " = " + variables[k] + ";\n}\n")
+
+if __name__=="__main__":
+
+    # A list of variables
+    variables = ["cdf", "stdID", "section", "y"]
+
+    # A list of datatypes 
+    datatypes = ["int", "float", "double" , "String"]
+
+    print_getter_setter(variables, datatypes)
+```

```

Comment:
@munenekelvin, kindly give an explanation of this program and how we got the output.